### PR TITLE
feat: expose shutdownmanager image to EnvoyGageway API

### PR DESCRIPTION
**Pull request by zhaohuabing**
This PR exposes shutdown manager image to EnvoyGageway API, so it can be populated via a helm value.

fix :https://github.com/envoyproxy/gateway/issues/3000 https://github.com/envoyproxy/gateway/issues/2808

### CODE REVIEW

File: internal/infrastructure/kubernetes/proxy/testdata/deployments/shutdown-manager.yaml [hunk 1]
```diff
@@ -249,7 +249,7 @@ spec:
             fieldRef:
               apiVersion: v1
               fieldPath: metadata.name
-        image: envoyproxy/gateway-dev:v1.2.3
+        image: privatereop/envoyproxy/gateway-dev:v1.2.3
```
Comment (arkodg) [inline internal/infrastructure/kubernetes/proxy/testdata/deployments/shutdown-manager.yaml:252]:
```suggestion
        image: privaterepo/envoyproxy/gateway-dev:v1.2.3
```

**Comment by arkodg** [comment]
thanks for adding this !
minor comment around typo, but this looks good !
as part of this fix, we need to make two more changes
* remove existing logic around shutdown versions using build flags https://github.com/envoyproxy/gateway/blob/9a626f262317b90946219d7f85069221cce2c4b6/internal/cmd/version/version.go#L24
cc @davidalger 
* by default, always make sure we set `EnvoyGateway.Shutdown.Image` in helm

is this something you want to tackle in this PR or follow up ?

**Comment by zhaohuabing** [comment]
> remove existing logic around shutdown versions using build flags
[gateway/internal/cmd/version/version.go](https://github.com/envoyproxy/gateway/blob/9a626f262317b90946219d7f85069221cce2c4b6/internal/cmd/version/version.go#L24)

The default image version is displayed in the `envoy-gateway version` command, as the `ENVOY_PROXY_VERSION` does. Or should we remove the version info for both the ShutdownManager and the Envoy Proxy?

File: tools/make/helm.mk [hunk 2]
```diff
@@ -33,4 +33,4 @@ helm-generate:
 
 helm-template: ## Template envoy gateway helm chart.
 	@$(LOG_TARGET)
-	helm template eg charts/gateway-helm --set deployment.envoyGateway.image.tag=latest > ./test/helm/default.yaml --namespace=envoy-gateway-system
+	helm template eg charts/gateway-helm --set deployment.envoyGateway.image.tag=latest --set config.envoyGateway.provider.kubernetes.shutdownManager.image="docker.io/envoyproxy/gateway-dev:latest" > ./test/helm/default.yaml --namespace=envoy-gateway-system
```
Comment (arkodg) [inline tools/make/helm.mk:36]: why do we need the extra `set` ?

**Review by arkodg — APPROVED** [review]
LGTM thanks !

**Review by shawnh2 — APPROVED** [review]
lgtm

**Review by cnvergence — APPROVED** [review]
🚀

File: tools/make/helm.mk (hunk 2 above)
Comment (zirain) [reply tools/make/helm.mk:36]: > | config.envoyGateway.provider.kubernetes.shutdownManager.image | string | `"docker.io/envoyproxy/gateway:latest"` | |

make this reasonable?

**Comment by wimnat** [comment]
When is this likely to get released?  I've just had to roll back to 0.6 because our k8s clusters have a gatekeeper rule that blocks images that are not from our private repo.  Without this fix to override the shutdown-manager image, I'm stuck :(

**Pull request merged**